### ISSUE TITLE
Feature request to improve the light bulb experience when multiple issues present in a line of code

### ISSUE THREAD

**Issue by morrisjoe** [dotnet/roslyn#5384]
Feature request from ALM MVP Terje Sandstrom

Current light bulb experience when multiple issues present in a line of code, where some have fixers and some don’t have fixers and some are hidden is confusing. When user does the quick action of ‘CTRL+.’, very confusing as to which one got fired. So this feature request is a proposal on how user experience can be presented in a way where there is no confusion

Scenario: A line of code has 2 code analysis issues present, where one rule involved is Code Cracker’s CC0073 and the other rule involved is Sonar Lint’S S131.  

Characteristics of the 2 involved rules are as follows…
• Code Cracker’s CC0073 is Hidden type. Rule is  ‘Add braces for each section in this switch’ and has a Fixer
• SonarLint’s S131 is Warning type. Rule is ‘Add a "default" clause to this switch statement’ and has no Fixer

Repro code is as follows. Enter the following lines of code.

``` C#
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ReproProblemTerje
{
    static class Program
    {
        enum Options
        {
            Option1,
            Option2,
            Option3
        }

        static void Main( )
        {
            NewMethod(Options.Option1);
        }

        private static void NewMethod(Options options)
        {

            switch (options)
            {
                case Options.Option1:

                        Do1();
                        break;

                case Options.Option2:

                        Do2();
                        break;

                case Options.Option3:

                        Do3();
                        break;

            }
        }

        private static void Do3()
        {
            throw new NotImplementedException();
        }

        private static void Do2()
        {
            throw new NotImplementedException();
        }

        private static void Do1()
        {
            throw new NotImplementedException();
        }
    }
}
```

The things that confuses the user for this scenario are…
1.  The rule without a code fix also show a “Show Potential Fixes”
2.  Only one rule is shown here, when there are two broken rules.

When one press the “Show potential Fixes”:
1.  The fix is for the other rule, which was not shown. 

Proposal to improve the user experience is as follows…

Regard a rule without a fix as a “strange case” and add it to the list of possible fixes, but with the message “No auto fix available”.  That would also put a certain pressure on the rule writers to provide fixes too.

Alternatively:  When there is a rule without a fix, don’t display the “Show potential fixes”.  Or make it obvious that the fixes are for something else.  

Likewise, it could be a thought to add some information regarding the hidden rules/fixes, so that the dev knows where it comes from. It could even say “Hidden rule/fix” or something to that effect.  
While on that:  In the fixes list there is no immediate information on which rule number it refers to, until you open up the fixes.  It would be nice to see that number as a prefix.

**Comment by Pilchie** [comment]
Example of the issue:

![image](https://cloud.githubusercontent.com/assets/312835/10080112/a9fd0666-62a2-11e5-80e8-85ca4dfb1099.png)

Clicking the "Show fixes" shows fixes for the other issue, you don't see:

![image](https://cloud.githubusercontent.com/assets/312835/10080134/c4655080-62a2-11e5-90a0-d331675c69e3.png)

---
Questions:
1. Why doesn't the suppress option show?  That would at least have *something* for the active item...

**Comment by Pilchie** [comment]
Let's think about options here @CyrusNajmabadi @DustinCampbell @dpoeschl

**Comment by Pilchie** [comment]
Also tagging @shyamnamboodiripad who might have thoughts.

**Comment by Pilchie** [comment]
We're going to have to continue thinking about this in 1.2.

**Comment by CyrusNajmabadi** [comment]
casey, any ideas here on a good experience?

**Comment by Pilchie** [comment]
Let's see if we hear anything after RC goes out.

**Comment by CyrusNajmabadi** [comment]
Closing out given lack of feedback.
